### DISCUSSION TITLE
Why does `magit-completing-read-multiple` not use `completing-read-multiple`?

### DISCUSSION THREAD

**Discussion by tsdh** [magit/magit#4406] [Support] [answered]
I use the [vertico](https://github.com/minad/vertico) completion frontend which advises the standard functions `completing-read-default` and `completing-read-multiple` in order to display its vertical, navigatable candidates list in the echo area.  This works fine for basically all completing reads I've encountered except for `magit-completing-read-multiple` which is used for example by `magit-log-other`, i.e., there I don't have the vertico UI but just the vanialla emacs completion.

Is there a striking reason why `magit-completing-read-multiple` doesn't use `completing-read-multiple`?  Its docstring says that, in contrast to the latter, it returns the input as string rather than as a list but `mapconcat`-ing the list could resembe that, so I guess there are other reasons why `completing-read-multiple` cannot be used here.

(Obviously, I hope that all those reasons can be overcome so that magit starts using  `completing-read-multiple`. :-))

**Comment by tarsius** [comment, marked as answer]
I've also recently switched to vertico and noticed this issue. And I mostly addressed it but haven't gotten around to finishing up the `crm` branch yet. One of the commit messages also answers some of your questions. I was in fact going to work on this again today. I might not merge this today, but it should be very soon.

**Reply by tsdh** [reply]
That's great to hear.  Thanks a lot!

**Reply by tarsius** [reply]
Took me longer than expected to get around to the polish. But now it's in #4420.

**Reply by tarsius** [reply]
Finally merged it!

**Comment by tsdh** [comment]
> Finally merged it!

Seems to work fine!

Thank you,
Tassilo
